anabeesh/src/main/java/com/cs18/anabeesh/ui/login/LoginActivity.java

```diff
@@ -44,9 +55,63 @@ public void onCreate(@Nullable Bundle savedInstanceState) {
 
     @OnClick(R.id.btn_login)
     void login() {
+
+        userEmail = emailEditText.getText().toString();
+        userPassword = passwordEditText.getText().toString();
+
+        if (userEmail.matches("")) {
```

Comment:
Ehh el talawos el ana shayfo da 😄, el logic ezay fl activity.

Plus badal mate3mel matches di estakhdem regular expression lel email, e3mel class esmaha textUtil msln feeha el validations di

E2ra f 7aga esmaha java regex

Reply:
- Done. 3mlt TextValidation Class , we 7teet feh isValidEmail b el regx
- 5leet matches tchof law empty field .

Reply:
Enta lw 3amalt add l da `(?=\S+$)` fel regex beta3ak msh haye2bal empty space khales, msh me7tag te3mel matches lel " ".

Plus lw 3ayez te3mel check 3ala 7aga empty estakhdem da `TextUtils.isEmpty(string)` di bete3mel check lw empty aw null w betragga3 boolean

Reply:
done 3mltha b 2edy gowa class TextValidation , gowa isValidEmail + isEmpty for any string